src/inc/til/some.h

```diff
@@ -127,6 +127,11 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
             return !_used;
         }
 
+        constexpr void clear() noexcept
+        {
+            _used = 0;

```

Comment:
A `std::array<N>` allocates all its space when it is constructed and doesn't free it until destructed. I'm also relatively certain that it will call the destructors on all members when it goes out of scope. The fact that we've hidden the members from the outside of the `til::some<T, N>` doesn't affect these rules for the embedded `std::array<N>`. It should free any elements, even the hidden ones, when it goes down as a part of the some destructor.

Reply:
I guess if you use `til::some<T, N>` with some sort of reference counted COM object, you're going to see it not get freed...

Reply:
K, I did what I think frees them.